geopandas/io/parquet.py

```diff
@@ -0,0 +1,228 @@
+import json
+
+from pandas.io.common import get_filepath_or_buffer
+from pandas.io.parquet import PyArrowImpl as PandasPyArrowImpl
+from pandas import DataFrame
+
+from geopandas.array import from_wkb, to_wkb
+from geopandas import GeoDataFrame
+import geopandas
+
+
+PARQUET_METADATA_VERSION = "1.0.0"
+
+
+def _encode_crs(crs):
+    if isinstance(crs, str):
+        return {"proj4": crs}
+
+    return crs
+
+
+def _decode_crs(crs):
+    if crs and "proj4" in crs:
+        return crs["proj4"]
+
+    return crs
+
+
+class PyArrowImpl(PandasPyArrowImpl):
+    """Extension of Pandas PyArrowImpl to handle serializing geometries to WKB.
+    """
+
+    def write(
+        self,
+        df,
+        path,
+        compression="snappy",
+        coerce_timestamps="ms",
+        index=None,
+        **kwargs
+    ):
+        """Write data frame to a parquet file, serializing any geometry
+        columns to WKB format.
+
+        See docstring for ``to_parquet``.
+        """
+
+        self.validate_dataframe(df)
+
+        geometry_columns = df.columns[df.dtypes == "geometry"].tolist()
+        geometry_column = df._geometry_column_name
+        crs = _encode_crs(df.crs)
+
+        # Convert to a DataFrame so we can convert geometries to WKB while
+        # retaining original column names
+        df = DataFrame(df.copy())
+
+        path, _, _, _ = get_filepath_or_buffer(path, mode="wb")
+
+        # Encode all geometry columns to WKB
+        for col in geometry_columns:
+            df[col] = to_wkb(df[col].values)
+
+        if index is None:
+            from_pandas_kwargs = {}
+        else:
+            from_pandas_kwargs = {"preserve_index": index}
+
+        table = self.api.Table.from_pandas(df, **from_pandas_kwargs)
+
+        metadata = table.schema.metadata
+
+        # Store geopandas specific metadata
+        metadata.update(
+            {
+                "geo": json.dumps(
+                    {
+                        "crs": crs,
+                        "primary_column": geometry_column,
+                        "columns": geometry_columns,
+                        "schema_version": PARQUET_METADATA_VERSION,
+                        "creator": {
+                            "library": "pyarrow",
```

Comment:
I think creator can be "geopandas" here, as I think it is mainly relevant which library created the metadata and made the WKB representation (that it is pyarrow that wrote file should be less relevant for the "geo" information, pyarrow also already includes that in the other metadata).